tests/simple_tests.rs

```diff
@@ -1878,6 +1878,103 @@ fn test_cwd_upper() {
     assert_eq!(vm.flags, X86Flags{..Default::default()});
 }
 
+#[test]
+fn test_pusha() {
+    let vm = execute_vm_with_asm("
+        mov ax, 0x6666
+        mov bx, 0x80
+        mov cx, 0xFF
+        mov dx, 0xFFFF
+        mov si, 0xFEDC
+        mov di, 0x6014
+        mov esp, 0x800065FE
+        mov bp, 0x6647
+        pushaw
+        hlt");
+    assert_eq!(vm.reg32(Reg32::ESP), 0x800065EE);
+    assert_eq!(vm.flags, X86Flags{..Default::default()});
+}
+
+#[test]
+fn test_popa() {
+    let vm = execute_vm_with_asm("
+        mov ax, 0x6666
+        mov bx, 0x80
+        mov cx, 0xFF
+        mov dx, 0xFFFF
+        mov si, 0xFEDC
+        mov di, 0x6014
+        mov esp, 0x800065FE
+        mov bp, 0x6647
+        pushaw
+        mov ax, 0
+        mov bx, 0
+        mov cx, 0
+        mov dx, 0
+        mov si, 0
+        mov di, 0
```

Comment:
This test should include setting the upper bits of the register (ie, EAX, EBX etc) to ensure that pushaw only saves the lower (16) bits and that popaw only restores the lower bits.

Reply:
done